x-pack/plugins/maps/public/classes/sources/es_search_source/util/scaling_documenation_popover.tsx

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Component, Fragment } from 'react';
+import { EuiButtonIcon, EuiPopover, EuiPopoverTitle, EuiText } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+
+interface Props {
+  limitOptionLabel: string;
+  clustersOptionLabel: string;
+  maxResultWindow: string;
+  mvtOptionLabel: string;
+}
+
+interface State {
+  isPopoverOpen: boolean;
+}
+
+export class ScalingDocumenationPopover extends Component<Props, State> {
+  state: State = {
+    isPopoverOpen: false,
+  };
+
+  _togglePopover = () => {
+    this.setState((prevState) => ({
+      isPopoverOpen: !prevState.isPopoverOpen,
+    }));
+  };
+
+  _closePopover = () => {
+    this.setState({
+      isPopoverOpen: false,
+    });
+  };
+
+  _renderContent() {
+    return (
+      <div>
+        <EuiPopoverTitle>
+          <FormattedMessage id="xpack.maps.scalingDocs.title" defaultMessage="Scaling" />
+        </EuiPopoverTitle>
+        <EuiText grow={false}>
+          <p>
+            <FormattedMessage
+              id="xpack.maps.scalingDocs.intro"
+              defaultMessage="Select the appropriate option for your use case."
+            />
+          </p>
+          <dl>
+            <dt>{this.props.limitOptionLabel}</dt>
+            <dd>
+              <p>
+                <FormattedMessage
+                  id="xpack.maps.scalingDocs.limitDetails"
+                  defaultMessage="Layer displays features from the first {maxResultWindow} documents. Results exceeding {maxResultWindow} are not displayed."
+                  values={{ maxResultWindow: this.props.maxResultWindow }}
+                />
+              </p>
+              <p>
+                <FormattedMessage
+                  id="xpack.maps.scalingDocs.limitUseCase"
+                  defaultMessage="Use this option to display medium data sets and create choropleth maps to compare statistics across boundaries."
+                />
+              </p>
+            </dd>
+
+            <dt>{this.props.clustersOptionLabel}</dt>
+            <dd>
+              <p>
+                <FormattedMessage
+                  id="xpack.maps.scalingDocs.clustersDetails"
+                  defaultMessage="When results exceed {maxResultWindow}, the layer uses GeoTile grid aggregation to group your documents into clusters and displays metrics for each cluster. When results are less then {maxResultWindow}, the layer displays features from individual documents."
```

Comment:
When results exceed 10,000, use a GeoTile a grid aggregation to group documents into clusters and display metrics for each cluster.  When less than 10,000, display features from individual documents.

Use this option for large data sets.  Does not support term joins.